doc/banner/banner_hld.md

```diff
@@ -0,0 +1,278 @@
+| 0.1 | 01/02/2023 | Sviatoslav Boichuk  | Initial version |
+
+
+# About this manual
+This document provides a high-level information for the SONiC Banner feature and Banner CLI. It describes high-level behavior, internal definition, design of the commands, syntax, and output definition.
+
+# Scope
+The scope of this document is to cover definition, design and implementation of SONiC Banner feature and Banner CLI.
+The document covers the next CLI:
+1.	Commands to configure banner messages
+2.	Command to display banner settings
+
+<!-- omit in toc -->
+## Abbreviations
+| Term  | Meaning                                   |
+|:------|:------------------------------------------|
+| SONiC | Software for Open Networking in the Cloud |
+| MOTD  | Message of the day                        |
+| DB    | Database                                  |
+| CLI   | Сommand-line Interface                    |
+| YANG  | Yet Another Next Generation               |
+
+<!-- omit in toc -->
+## List of figures
+- [Figure 1: Banner system chart diagram](#figure-1-banner-system-chart-diagram)
+- [Figure 2: Banner init flow](#figure-2-banner-init-flow)
+- [Figure 3: Banner config flow](#figure-3-banner-config-flow)
+- [Figure 4: Banner show configuration](#figure-4-banner-show-configuration)
+
+# 1 Introduction
+
+## 1.1 Feature overview
+The SONiC maintains several messages for communication with users. These messages are associated with login and logout processes.
+
+There are few banners message types:
+|  Type         |  Description                                                                    |
+| :-----------: | :-----------------------------------------------------------------------------: |
+|  login        | Display a banner to the users connected locally or remotely before login prompt |
+|  motd         | Display a banner to the users after login prompt                                |
+|  logout       | Display a logout banner to the users connected locally or remotely              |
+
+
+## 1.2 Requirements
+
+**This feature will support the following functionality:**
+1. Show Banner configuration
+2. Configure Banner
+    1. Feature state
+    2. Login message
+    3. Logout message
+    4. MOTD
+
+# 2 Design
+
+## 2.1 Overview
+Here is the representation of SONiC platform using Banner feature:
+
+![Banner system chart diagram](images/banner_system_chart_diagram.png "Figure 1: Banner system chart diagram")
+<!-- omit in toc -->
+###### Figure 1: Banner system chart diagram
+
+This feature require access to SONiC DB. All messages (MOTD, login and logout) saved into the SONiC config database. Hostcfgd will listen for the configuration changes in corresponding tables and restart banner-config service. Banner config service - it is simple SystemD service which runs before we get SSH connection. It reads configured messages from database and apply it to Linux.
+
+**The Linux files will be used:**
```

Comment:
I am wondering why should user not directly update the linux files. What is the value added by this HLD?

Is there a really scenarios that we need to config banner per device? If no, maybe a single command line to update these "Linux files" will be good enough. No dependency on database, hostcfgd, no need to have another service.